src/components/TodoItem.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import styles from './TodoItem.module.css';
+
+class TodoItem extends React.Component {
+  state = {
+    editing: false,
+  }
+
+  handleEditing = () => {
+    this.setState({
+      editing: true,
+    });
+  }
+
+  handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      this.setState({ editing: false });
+    }
+  }
+
+  render() {
+    const completedStyle = {
+      fontStyle: 'italic',
+      color: '#595959',
+      opacity: 0.4,
+      textDecoration: 'line-through',
+    };
+
+    // Destructuring todo
+    const { completed, id, title } = this.props.todo;
+
+    // for editing
+    const viewMode = {};
+    const editMode = {};
+
+    if (this.state.editing) {
+      viewMode.display = 'none';
+    } else {
+      editMode.display = 'none';
+    }
+
+    return (
+      <li className={styles.item}>
+        <div onDoubleClick={this.handleEditing} style={viewMode}>
+          <input
+            type="checkbox"
+            className={styles.checkbox}
+            checked={completed}
+            onChange={() => this.props.handleChangeProps(id)}
```

Comment:
To know the syntax for adding prop type validation, [this is a great](https://www.freecodecamp.org/learn/front-end-development-libraries/react/use-proptypes-to-define-the-props-you-expect) resource to check out.